tests/dragonfly/connection_test.py

```diff
@@ -1051,3 +1051,49 @@ async def test_hiredis(df_factory):
     server.start()
     client = base_redis.Redis(port=server.port, protocol=3, cache_config=CacheConfig())
     client.ping()
+
+
+@dfly_args({"proactor_threads": 1})
+async def test_pipeline_cache_size(df_factory):
+    server = df_factory.create(proactor_threads=1)
+    server.start()
+
+    # Start 1 client.
+    good_client = server.client()
+
+    await good_client.execute_command("set foo bar")
+
+    bad_actor_client = server.client()
+
+    info = await bad_actor_client.info()
+
+    # Cache is empty.
+    assert info["pipeline_cache_bytes"] == 0
+    assert info["dispatch_queue_bytes"] == 0
+
+    async def push_pipeline(bad_actor_client):
+        # Fill cache.
+        p = bad_actor_client.pipeline(transaction=True)
+        for i in range(1):
+            p.lpush(str(i), "V")
+        await p.execute()
+
+    await push_pipeline(bad_actor_client)
+    info = await good_client.info()
+
+    old_pipeline_cache_bytes = info["pipeline_cache_bytes"]
+    assert old_pipeline_cache_bytes > 0
+    assert info["dispatch_queue_bytes"] == 0
+
+    # Whoops, total pipeline_cache_bytes haven't changed. If a workload aggregates a bunch
+    # pipeline_cache_bytes because it recycled too many messages, they won't gradually be released
+    # if one command (one connection out of `n` connections) dispatches async. Only 1 command out of
+    # n connections must be dispatched async and the pipeline won't gradually be relesed.
+    for i in range(30):
```

Comment:
We cam drain the `pipeline cache bytes` once we stop dispatching async. But on large pool of connections only one command must dispatch async and then we need to internally reset the counter. If this pattern continues the size of the cache will remain constant and will not be released gradually.